README.md

```diff
@@ -1,10 +1,250 @@
+- [Add an A2A Java SDK Server Maven dependency to your project](#4-add-an-a2a-java-sdk-server-maven-dependency-to-your-project)
+
+### 1. Add the A2A Java SDK Core Maven dependency to your project
+
+```xml
+<dependency>
+    <groupId>io.a2a.sdk</groupId>
+    <artifactId>a2a-java-sdk-core</artifactId>
+    <version>${io.a2a.sdk.version}</version>
+</dependency>
+```
+
+### 2. Add a class that creates an A2A Agent Card
+
+```java
+import io.a2a.spec.AgentCapabilities;
+import io.a2a.spec.AgentCard;
+import io.a2a.spec.AgentSkill;
+import io.a2a.spec.PublicAgentCard;
+...
+
+@ApplicationScoped
+public class WeatherAgentCardProducer {
+    
+    @Produces
+    @PublicAgentCard
+    public AgentCard agentCard() {
+        return new AgentCard.Builder()
+                .name("Weather Agent")
+                .description("Helps with weather")
+                .url("http://localhost:10001")
+                .version("1.0.0")
+                .capabilities(new AgentCapabilities.Builder()
+                        .streaming(true)
+                        .pushNotifications(false)
+                        .stateTransitionHistory(false)
+                        .build())
+                .defaultInputModes(Collections.singletonList("text"))
+                .defaultOutputModes(Collections.singletonList("text"))
+                .skills(Collections.singletonList(new AgentSkill.Builder()
+                        .id("weather_search")
+                        .name("Search weather")
+                        .description("Helps with weather in city, or states")
+                        .tags(Collections.singletonList("weather"))
+                        .examples(List.of("weather in LA, CA"))
+                        .build()))
+                .build();
+    }
+}
+```
+
+### 3. Add a class that creates an A2A Agent Executor
+
+```java
+import io.a2a.server.agentexecution.AgentExecutor;
+import io.a2a.server.agentexecution.RequestContext;
+import io.a2a.server.events.EventQueue;
+import io.a2a.server.tasks.TaskUpdater;
+import io.a2a.spec.JSONRPCError;
+import io.a2a.spec.Message;
+import io.a2a.spec.Part;
+import io.a2a.spec.TextPart;
+...
+
+@ApplicationScoped
+public class WeatherAgentExecutorProducer {
+
+    @Inject
+    WeatherAgent weatherAgent;
+
+    // Thread pool for background execution
+    private final Executor taskExecutor = Executors.newCachedThreadPool();
+    
+    // Track active sessions for potential cancellation
+    private final ConcurrentHashMap<String, CompletableFuture<Void>> activeSessions = new ConcurrentHashMap<>();
+
+    @Produces
+    public AgentExecutor agentExecutor() {
+        return new WeatherAgentExecutor(weatherAgent, taskExecutor, activeSessions);
+    }
+
+    private static class WeatherAgentExecutor implements AgentExecutor {
+
+        private final WeatherAgent weatherAgent;
+        private final Executor taskExecutor;
```

Comment:
Is this and the activeSessions needed? We removed them from the TCK a while back